tools/distpartitioning/parmetis_preprocess.py

```diff
@@ -0,0 +1,326 @@
+    if constants.STR_NODE_DATA in schema_map:
+        dataset_features = schema_map[constants.STR_NODE_DATA]
+        if((dataset_features is not None) and (len(dataset_features) > 0)):
+            for ntype_name, ntype_feature_data in dataset_features.items():
+                if ntype_name != tgt_ntype_name : 
+                    continue
+                # ntype_feature_data is a dictionary
+                # where key: feature_name, value: dictionary in which keys are "format", "data".
+                for feat_name, feat_data in ntype_feature_data.items():
+                    if feat_name in feat_names:
+                        my_feat_data_fname = feat_data[constants.STR_DATA][rank]
+                        logging.info(f'Reading: {my_feat_data_fname}')
+                        if (os.path.isabs(my_feat_data_fname)):
+                            node_features[feat_name] = np.load(my_feat_data_fname)
+                        else:
+                            node_features[feat_name] = np.load(os.path.join(input_dir, my_feat_data_fname))
+    return node_features
+
+
+def gen_node_weights_files(schema_map, output):
+    """
+    Function to create node weight files for ParMETIS along with the edge files.
+
+    This function generates node-data files, which will be read by the ParMETIS
+    executable for partitioning purposes. Each line in these files will be of the 
+    following format: 
+    <node_type_id> <weight_list> <type_node_id>
+    where 
+    node_type_id -  is id assigned to the node-type to which a given particular
+    node belongs to
+    weight_list - this is a one-hot vector in which the number in the location of
+    the current nodes' node-type will be set to `1` and other will be `0`
+    type_node_id - this is the id assigned to the node (in the context of the current
+    nodes` node-type). Meaning this id is unique across all the nodes which belong to 
+    the current nodes` node-type.
+
+    Parameters:
+    -----------
+    schema_map : json dictionary
+        dictionary created by reading the metadata.json file for the input dataset
+    output : string
+        location of storing the node-weights and edge files for ParMETIS
+
+    Returns:
+    --------
+    list : 
+        list of filenames for nodes of the input graph
+    list : 
+        list o ffilenames for edges of the input graph
+    """
+    rank = get_proc_info()
+    ntypes_ntypeid_map, ntypes, ntid_ntype_map = get_node_types(schema_map)
+    type_nid_dict, ntype_gnid_offset = get_idranges(schema_map[constants.STR_NODE_TYPE],
+                                        schema_map[constants.STR_NUM_NODES_PER_CHUNK])
+
+    node_files = []
+    outdir = Path(output)
+    os.makedirs(outdir, exist_ok=True)
+
+    for ntype_id, ntype_name in ntid_ntype_map.items():
+        type_start, type_end = type_nid_dict[ntype_name][rank][0], type_nid_dict[ntype_name][rank][1]
+        count = type_end - type_start
+        sz = (count,)
```

Comment:
What is sz? let's be careful when we invent new abbr. If this is a convention in neural network, please ignore my comment.